src/lib/sql/sql_translator.cpp

```diff
@@ -266,6 +277,26 @@ std::shared_ptr<AbstractLQPNode> SQLTranslator::_translate_select_statement(cons
   return _current_lqp;
 }
 
+void SQLTranslator::_translate_hsql_with_description(hsql::WithDescription& desc) {
+  SQLTranslator with_translator{_use_mvcc, nullptr, _parameter_id_allocator, _with_descriptions};
+  auto lqp = with_translator._translate_select_statement(*desc.select);
+
+  // Save mappings: ColumnID -> ColumnName
+  std::unordered_map<ColumnID, std::string> column_names;
+  for (auto column_id = ColumnID{0}; column_id < lqp->column_expressions().size(); ++column_id) {
+    const auto identifiers =
+        with_translator._sql_identifier_resolver->get_expression_identifiers(lqp->column_expressions()[column_id]);
+    for (const auto& identifier : identifiers) {
+      column_names.insert_or_assign(column_id, identifier.column_name);
```

Comment:
This is something I want to discuss with you @mrzzzrm today.

Reply:
Has to be tomorrow, sorry. Today I won't be in

Reply:
So basically you're invalidating all but one name of the column, right? How did you handle this for views @janetzki ?

Reply:
Good point. I did not make any changes in `_translate_create_view()` so far. I'll update you as soon as I have tested `CREATE VIEW` for hyrise-mp/hyrise#38.

Reply:
Did you update LQPView yet? @janetzki

Reply:
Yes, in https://github.com/hyrise/hyrise/pull/1692/commits/a926f691958e8cd2b2ffca648843d8275dee2eb5.